### ISSUE TITLE
PostgreSQL \crosstabview still broken in some cases

### ISSUE THREAD

**Issue by pgr0ss** [sqlfluff/sqlfluff#7876]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The previous issue (#7815) was closed, but there are other cases which are not working:

Here's a more complicated example:

`crosstab2.sql`:
```sql
select
    a,
    b,
    c
from mytable
order by 1
\crosstabview

with entries as (select 1, 2, 3)
select
    a,
    b,
    c
from entries
order by 1
\crosstabview
```

```bash
> uvx sqlfluff@4.2.1 format --dialect postgres crosstab2.sql
==== finding fixable violations ====
== [crosstab2.sql] FAIL
  [1 templating/parsing errors found]
  == [crosstab2.sql] ==
  L:  20 | P:   1 |  PRS | Line 20, Position 1: Found unparsable section:
                       | '\\crosstabview'
==== no fixable linting violations found ====
All Finished 📜 🎉!
  [6 unfixable linting violations found]
```

### Expected Behaviour

I expected the file to parse and format.

### Observed Behaviour

See above

### How to reproduce

See above

### Dialect

postgres

### Version

4.2.1

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by IdanGonen** [comment]
@pgr0ss  @peterbud 
[PR Link](https://github.com/sqlfluff/sqlfluff/pull/7895)
